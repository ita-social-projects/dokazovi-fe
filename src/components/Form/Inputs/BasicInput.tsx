import { IconButton, InputAdornment, TextField } from '@material-ui/core';
import { Visibility, VisibilityOff } from '@material-ui/icons';
import clsx from 'clsx';
import React, { useState } from 'react';
import { useStyles } from './styles/BasicInput.style';

interface IBasicInputProps {
  className?: string;
  name?: string;
  inputRef?: React.Ref<any>;
  label?: string;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  error?: boolean;
  helperText?: React.ReactNode;
  type?: 'text' | 'password';
}

export const BasicInput: React.FC<IBasicInputProps> = ({
  className,
  inputRef,
  name,
  label,
  onChange,
  error,
  helperText,
  type = 'text',
}) => {
  const classes = useStyles();
  const [showPassword, setShowPassword] = useState(false);
  return (
    <TextField
      className={clsx(classes.basicInput, className)}
      type={type === 'text' ? type : showPassword ? 'text' : 'password'}
      variant="outlined"
      margin="normal"
      fullWidth
      name={name}
      inputRef={inputRef}
      label={label}
      onChange={onChange}
      error={error}
      helperText={helperText}
      InputProps={{
        endAdornment: type === 'password' && (
          <InputAdornment position="end">
            <IconButton
              aria-label="toggle password visibility"
              className={classes.visibilityIconButton}
              onClick={() => setShowPassword(!showPassword)}
            >
              {showPassword ? (
                <VisibilityOff className={classes.visibilityIcon} />
              ) : (
                <Visibility className={classes.visibilityIcon} />
              )}
            </IconButton>
          </InputAdornment>
        ),
      }}
    />
  );
};
