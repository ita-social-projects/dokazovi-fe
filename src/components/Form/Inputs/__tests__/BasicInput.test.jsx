import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BasicInput } from '../BasicInput';

jest.mock('axios');
jest.mock('request');

describe('BasicInput component tests', () => {
  it('should render BasicInput component', () => {
    const { asFragment } = render(<BasicInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render password type input', () => {
    const onChange = jest.fn();
    const typeMock = {
      type: 'password',
    };
    render(<BasicInput type={typeMock.type} onChange={onChange} />);

    userEvent.type(screen.getByTestId('basic-input'), 'mchsye34');

    expect(onChange).toHaveBeenCalledTimes(8);
  });

  it('should toggle visibility icon', async () => {
    const typeMock = {
      type: 'password',
    };
    render(<BasicInput type={typeMock.type} />);

    userEvent.type(screen.getByTestId('basic-input'), 'mchsye34');
    userEvent.click(screen.getByLabelText('toggle password visibility'));

    await waitFor(() =>
      expect(screen.getByTestId('visibility-off')).toBeInTheDocument(),
    );

    userEvent.click(screen.getByLabelText('toggle password visibility'));

    await waitFor(() =>
      expect(screen.getByTestId('visibility')).toBeInTheDocument(),
    );
  });
});
