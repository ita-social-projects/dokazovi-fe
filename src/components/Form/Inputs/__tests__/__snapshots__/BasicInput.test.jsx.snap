// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInput component tests should render BasicInput component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root BasicInput-basicInput-1 MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        data-testid="basic-input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-5"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
